src/oslogin_utils.cc

```diff
@@ -937,6 +937,50 @@ bool ParseJsonToChallenges(const string& json, std::vector<Challenge>* challenge
   return ret;
 }
 
+bool ParseJsonToCAKeys(const string& json, std::vector<string>& ca_keys) {
+  json_object* root = NULL;
+  root = json_tokener_parse(json.c_str());
+  if (root == NULL) {
+    return false;
+  }
+
+  json_object* trusted_cas = NULL;
+  if (!json_object_object_get_ex(root, "trustedCertificateAuthorities", &trusted_cas)) {
+    goto fail;
+  }
+  if (json_object_get_type(trusted_cas) != json_type_array) {
+    goto fail;
+  }
+
+  {
```

Comment:
I'm glad you asked me about that tricky part! ;)
You're right, I didn't mean to use `if` in there, as there is no missing condition. 
The reason for using a new block is to create a new scope. Before the block, there are a couple if statements which `goto` to the clean up part at the end of the function. At the same time, below those goto statements there is  basically a loop for iterating an array (and only if that json key exists as an array). So, a few additional variables are declared. But, because a compiler sees goto before a declaration of a few variables, it fails to proceed. The compiler doesn't know where in the code it will be jumped so it is risky to bypass declarations. A new scope block helps to fix this issue.

For example, this code
```
  1 #include <iostream>
  2 
  3 int main() {
  4   int i = 1;
  5   std::cout << i << std::endl;
  6   {
  7     int i = 2; 
  8     std::cout << i << std::endl;
  9   
 10   std::cout << i << std::endl;
 11 }
 12   return 0;
 13 }
```
It prints out: `1 2 1`
But, if I remove `int` in the line 7, then it will  print `1 2 2`. If I remove that inner block, then a compiler will fail as there is a duplicate declaration.

This technique is pretty useful if you are not sure about using some variable withing the scope and to prevent overriding it, a new scope with declaration is coming handy. 
 
Here is another cool example: https://www.fayewilliams.com/2011/10/26/declaring-variables-in-switch-statements/

Reply:
Ah, interesting that the compiler actually complains about skipping past those variable declarations; I had assumed that it would have allocated stack space for all variables (regardless of where they are actually declared in code) at the top of the stack frame, C-style, then just assigned them values at the point in code where they are "declared". 

That's neat, that it actually obeys the code more directly and will complain if it doesn't make sense.

/lgtm